contrib/epee/include/net/net_utils_base.h

```diff
@@ -94,17 +94,51 @@ namespace net_utils
 		END_KV_SERIALIZE_MAP()
 	};
 
-	inline bool operator==(const ipv4_network_address& lhs, const ipv4_network_address& rhs) noexcept
+	class ipv4_network_subnet
+	{
+		uint32_t m_ip;
+		uint8_t m_mask;
+
+	public:
+		constexpr ipv4_network_subnet() noexcept
+			: ipv4_network_subnet(0, 0)
+		{}
+
+		constexpr ipv4_network_subnet(uint32_t ip, uint8_t mask) noexcept
+			: m_ip(ip), m_mask(mask) {}
+
+		bool equal(const ipv4_network_subnet& other) const noexcept;
+		bool less(const ipv4_network_subnet& other) const noexcept;
+		constexpr bool is_same_host(const ipv4_network_subnet& other) const noexcept
+		{ return ip() == other.ip(); }
+                bool matches(const ipv4_network_address &address) const;
+
+		constexpr uint32_t ip() const noexcept { return m_ip & ~(0xffffffffull << m_mask); }
```

Comment:
Does this function make sense? Its technically not a valid IP most often.

Reply:
OK, I'll rename it.